chain/client/src/sync/block.rs

```diff
@@ -228,36 +247,33 @@ impl BlockSync {
             if let Some(peer) = peer {
                 debug!(
                     target: "sync",
-                    head_height = chain_head.height,
-                    header_head_height = header_head.height,
-                    block_hash = ?hash,
-                    block_height = height,
+                    block_hash = ?next_hash,
+                    block_height = next_height,
                     request_from_archival,
                     peer = ?peer.peer_info.id,
                     num_peers = highest_height_peers.len(),
-                    "Block sync: requested block"
+                    "requested block"
                 );
                 self.network_adapter.send(PeerManagerMessageRequest::NetworkRequests(
-                    NetworkRequests::BlockRequest { hash, peer_id: peer.peer_info.id.clone() },
+                    NetworkRequests::BlockRequest {
+                        hash: next_hash,
+                        peer_id: peer.peer_info.id.clone(),
+                    },
                 ));
                 num_requests += 1;
             } else {
                 warn!(
                     target: "sync",
-                    head_height = chain_head.height,
-                    header_head_height = header_head.height,
-                    block_hash = ?hash,
-                    block_height = height,
+                    block_hash = ?next_hash,
+                    block_height = next_height,
                     request_from_archival,
-                    "Block sync: No available peers to request a block from");
+                    "no available peers to request a block from");
             }
         }
         debug!(
             target: "sync",
-            head_height = chain_head.height,
-            header_head_height = header_head.height,
             num_requests,
-            "Block sync: requested blocks");
+            "requested blocks");
```

Comment:
This could also be a field on the span, right? Though I guess if we don't print any events then span data won't be easily visible without logging out span lifetime events…

Reply:
I went further and introduced `BlockKnowledge` enum to use instead of `Result<(), BlockKnownError>`: c9e88fb3a